File: Exercise/Very Hard/10. Casino Security/Code/Casino Security.js
function casinoSecurity(s) {
    let thiefPos = s.indexOf('T');
    let moneyPos = s.indexOf('$');
    let guardPos = s.indexOf('G');

    if (thiefPos === -1 || moneyPos === -1) {
        return "Safe";
    }

    if ((thiefPos < moneyPos && moneyPos < guardPos) ||
        (guardPos < moneyPos && moneyPos < thiefPos)) {
        return "Safe";
    }

    return "ALARM!";
}
let input = "xxxxTTxGxx$xxTxxx";

let result = casinoSecurity(input);
console.log(result);

File: Exercise/Very Hard/10. Casino Security/Code/Casino Security.cs
using System;

class Program {
    static string CasinoSecurity(string s) {
        int thiefPos = s.IndexOf('T');
        int moneyPos = s.IndexOf('$');
        int guardPos = s.IndexOf('G');

        if (thiefPos == -1 || moneyPos == -1) {
            return "Safe";
        }

        if ((thiefPos < moneyPos && moneyPos < guardPos) ||
            (guardPos < moneyPos && moneyPos < thiefPos))
        {
            return "Safe";
        }

        return "ALARM!";
    }

    static void Main() {
        string input = Console.ReadLine();

        string result = CasinoSecurity(input);
        Console.WriteLine(result);
    }
}



